documentation/src/main/asciidoc/topics/con_cache_encoding.adoc

```diff
@@ -0,0 +1,40 @@
+[id='encoding']
+= Cache encoding
+Encoding is the format, identified by a media type, that {brandname} uses to store entries (key/value pairs) in caches.
+
+[discrete]
+== Remote caches
+
+{brandname} Server stores entries in remote caches with the encoding that is set in the cache configuration.
+If the remote cache does not have any encoding configuration, {brandname} Server stores keys and values without any media type information.
+Hot Rod and REST clients also use encoding to operate on data and include a media type with each request to {brandname} Server.
+
+If you do not configure encoding for remote caches then {brandname} Server stores keys and values exactly as the client sends them, discarding any media type information.
+When clients then request entries from these caches, {brandname} Server returns keys and values as they are stored in the remote cache and ignores the client's media type.
+When the remote cache has an encoding, other than `application/unknown`, {brandname} Server converts keys and values to and from the client media type with each request.
```

Comment:
We can condense a bit this section at the same time introducing the basic data conversion mechanics, and getting rid of APPLICATION_UNKWOWN, which is a deprecated hack for backwards compatibility reasons:

```suggestion
{brandname} Server stores entries in remote caches with the encoding that is set in the cache configuration.

Hot Rod and REST clients include a media type with each request to {brandname} Server; those clients can request data in a different format from storage and the server will do a conversion on-demand if it supports it, otherwise an error is generated. Similarly, when those clients write data to the server in a different format than configured, {brandname} will convert the data the storage format. If the server cannot convert the incoming data to its storage format, an error is also generated.

If you do not configure encoding for remote caches then {brandname} Server assumes keys and values have unknown format, and conversions are done assuming data as generic ```byte[]``` with unknown format.
```

Reply:
@gustavonalle Hey, just to make sure we don't lose anything - and to keep the review going - I've issued https://github.com/infinispan/infinispan/pull/9327 that includes this comment. Thanks.